cbor/record.go

```diff
@@ -0,0 +1,206 @@
+package cbor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-ipld-cbor"
+	"github.com/ipfs/go-ipld-format"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/textileio/go-textile-core/crypto"
+	"github.com/textileio/go-textile-core/thread"
+	pb "github.com/textileio/go-textile-threads/pb"
+)
+
+func init() {
+	cbornode.RegisterCborType(record{})
+}
+
+type record struct {
+	Block cid.Cid
+	Sig   []byte
+	Prev  cid.Cid `refmt:",omitempty"`
+}
+
+func NewRecord(ctx context.Context, dag format.DAGService, block format.Node, prev cid.Cid, sk ic.PrivKey, key crypto.EncryptionKey) (thread.Record, error) {
```

Comment:
Records are the top-most object in a log. The `block` is an event in our case, but I've left it generic so any node can be wrapped.

Reply:
just to make sure i follow, what's a block node that isn't an event? how is it used?

Reply:
it could be anything... up to the user. It's a little half-baked, but the idea is that somebody could use the bulk of this lib for something else that needs append-only single writer logs that aren't of the same shape as our events.